Reverting changes in 03d1c4666d701283ceba479174925963fa607f3b but with improvements

See https://github.com/pimutils/vdirsyncer/pull/1204 for bumping the required setuptools to actually work with the new declarations
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,11 +16,11 @@ description = "Synchronize calendars and
 readme = "README.rst"
 requires-python = ">=3.8"
 keywords = ["todo", "task", "icalendar", "cli"]
-license = "BSD-3-Clause"
-license-files = ["LICENSE"]
+license = {file = "LICENSE"}
 classifiers = [
     "Development Status :: 4 - Beta",
     "Environment :: Console",
+    "License :: OSI Approved :: BSD License",
     "Operating System :: POSIX",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3.10",
